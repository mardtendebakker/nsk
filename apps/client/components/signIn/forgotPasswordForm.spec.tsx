import { fireEvent, render, waitFor } from '@testing-library/react';
import ForgotPasswordForm from './forgotPasswordForm';

const mockUseSecurity = {
  state: { get user() { return null; }, get loading() { return false; } },
  forgotPassword: jest.fn(),
};

jest.mock('../../hooks/useSecurity', () => jest.fn(() => mockUseSecurity));

describe('forgotPasswordForm', () => {
  afterEach(() => {
    mockUseSecurity.forgotPassword.mockReset();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<ForgotPasswordForm onFormSelect={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('calls onFormSelect with correct value', () => {
    const onFormSelect = jest.fn();
    const { getByText } = render(<ForgotPasswordForm onFormSelect={onFormSelect} />);

    fireEvent.click(getByText('signIn'));
    expect(onFormSelect).toBeCalledWith({ form: 'signIn' });
  });

  it("doesn't submit if form is invalid", () => {
    const { container } = render(<ForgotPasswordForm onFormSelect={() => {}} />);

    fireEvent.submit(container.querySelector('form[name="forgotPassword"]'));
    expect(mockUseSecurity.forgotPassword).not.toBeCalled();
  });

  it('submits if form is valid', () => {
    const onFormSelect = jest.fn();
    const { container } = render(<ForgotPasswordForm onFormSelect={onFormSelect} />);
    const emailOrUsername = 'username';

    const emailOrUsernameInput = container.querySelector("input[name='emailOrUsername']");
    fireEvent.change(emailOrUsernameInput, { target: { value: emailOrUsername } });

    fireEvent.submit(container.querySelector('form[name="forgotPassword"]'));
    expect(mockUseSecurity.forgotPassword).toBeCalledWith({ emailOrUsername });
    waitFor(() => expect(onFormSelect).toHaveBeenCalledWith({ form: 'changePassword' }));
  });

  it('performs no action when loading', () => {
    const onFormSelect = jest.fn();
    jest.spyOn(mockUseSecurity.state, 'loading', 'get').mockReturnValue(true);

    const { container, getByText } = render(<ForgotPasswordForm onFormSelect={onFormSelect} />);
    const emailOrUsername = 'username';

    const emailOrUsernameInput = container.querySelector("input[name='emailOrUsername']");
    fireEvent.change(emailOrUsernameInput, { target: { value: emailOrUsername } });

    fireEvent.submit(container.querySelector('form[name="forgotPassword"]'));

    fireEvent.click(getByText('signIn'));

    const submitButton = container.querySelector("button[type='submit']");
    fireEvent.click(submitButton);

    expect(onFormSelect).not.toBeCalled();
    expect(mockUseSecurity.forgotPassword).not.toBeCalled();
  });
});
