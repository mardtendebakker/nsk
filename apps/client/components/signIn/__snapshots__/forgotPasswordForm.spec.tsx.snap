// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`forgotPasswordForm matches snapshot 1`] = `
<DocumentFragment>
  <h4
    class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1vw6mcs-MuiTypography-root"
  >
    forgotPasswordMessage
  </h4>
  <div
    class="MuiStack-root css-duiheb-MuiStack-root"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5 css-50uozi-MuiTypography-root"
    >
      signIn
    </h5>
  </div>
  <form
    name="forgotPassword"
  >
    <div
      class="MuiStack-root css-1pxjrel-MuiStack-root"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-y72rub-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1gx9giq-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for=":r0:"
          id=":r0:-label"
        >
          emailOrUsername
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r0:"
            name="emailOrUsername"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-14lo706"
            >
              <span>
                emailOrUsername
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary MuiButton-fullWidth css-1evprfk-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="submit"
    >
      confirm
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </form>
</DocumentFragment>
`;
